.github/workflows/ci.yml

```diff
@@ -131,14 +131,194 @@ jobs:
       run: |
         tox -e py
 
-  verify-desktop:
-    name: Desktop app verification
-    needs: smoke
+  verify-desktop-macOS:
+    name: macOS desktop app verification
+    needs: platform-compat
```

Comment:
Is it necessary to have 5 configurations which have so much in common? Could the differences (e.g. cache directories, `--adhoc-sign` argument) and matrix limitations (e.g. building iOS on macOS only) have been dealt with using `include` variables and `if`?

Reply:
I definitely agree this is... verbose. The original version of this stage was a matrix build; however, adding iOS and Android as another axis, while only building Toga got prohibitively complex. However, thinking about it again, it might be easier to do this as a single OS x project matrix, and then exclude iOS and Android based on a single axis of the matrix. I'll have a fiddle and see what I can do.

Reply:
Ok - I've been able to clean this up quite a bit. It turns out it was the use of `tox` that was making the process complex; once I dropped Tox, the matrix and step definitions actually ended up quite elegant.